lib/avalon/batch/entry.rb

```diff
@@ -277,15 +285,31 @@ def self.caption_language(language)
       end
       private_class_method :caption_language
 
+      def self.process_datastream(datastream, type)
+        file, label, language = ["_file", "_label", "_language"].map { |item| item.prepend(type).to_sym }
```

Comment:
What do you think about renaming these variables so they are distinct from the values stored on lines 293-295?
```suggestion
        file_key, label_key, language_key = ["_file", "_label", "_language"].map { |item| item.prepend(type).to_sym }
```